docs/DeveloperGuide.md

```diff
@@ -1,38 +1,1061 @@
+
+__Communication between `Frontend` and `Backend`__
+- PocketPal uses the `Gson` library to serialise and deserialise objects where applicable, such as `Entry` and `EntryLog`.
+- This allows us to standardise the communication framework between `Frontend` and `Backend`, by using a simplified HTTP model.
+- Each endpoint takes in a `Request`, and returns a `Response` based on the requested data.
+
+## Frontend
+
+<!-- @@author adenteo -->
+### Parser
+
+The `Parser` class is a fundamental component instantiated as soon as PocketPal is initialised. Its __purpose__ is to
+convert the user's input into structured data which then produces clear instructions for the rest of the program.
+
+Some of its core features include:
+
+- Breaking down user input and extracting the relevant data for further processing.
+- Performing input validation and error handling to ensure that input data is in the correct format and ready to be
+  processed.
+- Converting the input data into the correct format and returning it as a `Command` class to be further processed by the
+  application.
+
+Here's a class diagram that shows the core structure of the `Parser` class.
+
+![ParserClassDiagram](static/frontend/parser/ParserClassDiagram.png)
+
+How `Parser` works:
+
+1. When a user enters a command, the `Frontend` uses `Parser` to resolve the user input via `parseUserInput()`.
+2. Within `parseUserInput()`, the corresponding `parseXCommand()` (`X` is a placeholder for the various command
+   names[^1] e.g. `parseAddCommand()`, `parseDeleteCommand()`.)  is invoked to validate that the user input is in the
+   correct format. Any exceptions will be thrown and their corresponding error messages will be shown to the user via
+   the `ui` class.
+2. If the user input is valid, an `XCommand` object containing the relevant data is created and returned.
+   E.g. `parseAddCommand()` would create a `AddCommand` object containing the description, price and category.
+3. From there, the `XCommand` is ready to be executed by the program. (All `XCommand` classes inherit from `Command` and
+   have corresponding `execute()` that carry out their specific instructions.)
+
+[^1]: A list of currently supported commands in PocketPal can be found [here](../../UserGuide.html/features/)
+
+The Sequence Diagram below illustrates the interactions within the `Parser` component when a user inputs the following
+command: `/add McDonalds -c Food -p 10.50`
+
+![ParserSequenceDiagram](static/frontend/parser/ParserSequenceDiagram.png)
+
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div>
+<!-- @@author -->
+<!-- @@author kaceycsn -->
+
+### Commands
+
+#### Add Command
+The add entry mechanism is facilitated by `EntryLog`. Every instance of `AddCommand` is created with an `Entry` instance.
+
+The following sequence diagram shows how the add command work:
+
+![AddCommandSequenceDiagram](static/frontend/commands/AddCommandSequenceDiagram.png)
```

Comment:
For the instance postReq:Request, there is not method calling it. Maybe can add the method name invoking the instance together with the arrow.